src/gluonts/nursery/duru/hyperparams.py

```diff
@@ -0,0 +1,623 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from utils import str2bool
+import wandb
+import os
+import pickle5 as pickle  # on Python 3.6 must use this
+import argparse
+
+
+class Hyperparams(dict):
+    """
+    Wrapper class for a dictionary required for pickling.
+    """
```

Comment:
Not sure I understand the need for this.

Reply:
It's required in order to pickle it correctly. There was an issue that I don't fully recall if we don't do this, probably due to the diversity of types the dictionary is pointing to.

Reply:
If you know the types of the variables, it's probably easier to use something like `pydantic` to define which fields are there.

Reply:
this is nice. For simplicity, I would suggest to leave it as is for now

Reply:
There is also types.SimpleNamespace which should also do the trick.